PyInstaller/depend/utils.py

```diff
@@ -119,121 +119,115 @@ def scan_code_for_ctypes(co):
-        #
-        #  ctypes.util.find_library('gs')
-        #
-        #     LOAD_GLOBAL   0 (ctypes)
-        #     LOAD_ATTR     1 (util) <--- we "are" here right now
-        #     LOAD_ATTR     1 (find_library)
-        #     LOAD_CONST    1 ('gs')
-        instruction = next(instructions)
-        if instruction.opname == 'LOAD_ATTR':
-            if co.co_names[instruction.arg] == "find_library":
-                libname = _libFromConst()
-                if libname:
-                    lib = ctypes.util.find_library(libname)
-                    if lib:
-                        # On Windows, `find_library` may return
-                        # a full pathname. See issue #1934
-                        return os.path.basename(lib)
+            if not instruction or instruction.opname not in expected_ops:
+                continue
+
+            name = instruction.argval
+            if name == "ctypes":
+                # Guesses ctypes has been imported as `import ctypes` and
+                # the members are accessed like: ctypes.CDLL("library.so")
+                #
+                #   LOAD_GLOBAL 0 (ctypes) <--- we "are" here right now
+                #   LOAD_ATTR 1 (CDLL)
+                #   LOAD_CONST 1 ('library.so')
+                #
+                # In this case "strip" the `ctypes` by advancing and expecting
+                # `LOAD_ATTR` next.
+                expected_ops = ('LOAD_ATTR',)
+                instruction = next(instructions)
+                if instruction.opname not in expected_ops:
+                    continue
+                name = instruction.argval
+
+            if name in ("CDLL", "WinDLL", "OleDLL", "PyDLL"):
+                # Guesses ctypes imports of this type: CDLL("library.so")
+                #
+                #   LOAD_GLOBAL 0 (CDLL) <--- we "are" here right now
+                #   LOAD_CONST 1 ('library.so')
+
+                yield _libFromConst()
+
+            elif name in ("cdll", "windll", "oledll", "pydll"):
+                # Guesses ctypes imports of these types:
+                #
+                #  * cdll.library (only valid on Windows)
+                #
+                #     LOAD_GLOBAL 0 (cdll) <--- we "are" here right now
+                #     LOAD_ATTR 1 (library)
+                #
+                #  * cdll.LoadLibrary("library.so")
+                #
+                #     LOAD_GLOBAL   0 (cdll) <--- we "are" here right now
+                #     LOAD_ATTR     1 (LoadLibrary)
+                #     LOAD_CONST    1 ('library.so')
+                instruction = next(instructions)
+                if instruction.opname == 'LOAD_ATTR':
+                    if instruction.argval == "LoadLibrary":
+                        # Second type, needs to fetch one more instruction
+                        yield _libFromConst()
+                    else:
+                        # First type
+                        yield instruction.argval + ".dll"
+
```

Comment:
Okay, you convinced me. Adding support for `.argval` would add quite a few lines of code.